quinn/src/endpoint.rs

```diff
@@ -342,8 +349,17 @@ impl EndpointInner {
             });
         let mut iovs = unsafe { iovs.assume_init() };
         loop {
-            match self.socket.poll_recv(cx, &mut iovs, &mut metas) {
-                Poll::Ready(Ok(msgs)) => {
+            match self.socket.poll_recv_ready(cx) {
+                Poll::Pending => break,
+                Poll::Ready(Err(e)) => return Err(e),
+                Poll::Ready(Ok(())) => {}
+            }
+            let socket_state = &mut self.socket_state;
+            let socket = &self.socket;
+            match socket.try_io(Interest::READABLE, || {
```

Comment:
Style nit: would like to avoid multi-line scrutinees.

Thought, I think 2021 edition is now in scope, so can we avoid all the borrow splitting before passing state into a closure?

Reply:
Oh yeah, forgot that was a thing now. Fixed.